bindsnet/datasets/alov300.py

```diff
@@ -0,0 +1,318 @@
+from typing import Optional, Tuple, List, Iterable
+from urllib.request import urlretrieve
+from torchvision import transforms
+import xml.etree.ElementTree as ET
+from torch.utils.data import Dataset
+
+warnings.filterwarnings("ignore")
+
+
+class ALOV300(torch.utils.data.Dataset):
+    DATASET_WEB = "http://alov300pp.joomlafree.it/dataset-resources.html"
+    VOID_LABEL = 255
+
+    def __init__(
+        self,
+        root,
+        transform,
+        input_size,
+        codalab=False,
+        download=False,
+        num_samples: int = -1,
+    ):
+        """
+        Class to read the ALOV dataset
+        
+        :param root: Path to the ALOV folder that contains JPEGImages, Annotations, etc. folders.
+        :param input_size: The input size of network that is using this data, for rescaling
+        :param download: Specify whether to download the dataset if it is not present
+        :param num_samples: Number of samples to pass to the batch
+        """
+        super(ALOV300, self).__init__()
+
+        # Makes a unique path for a given instance of davis
+        self.root = root
+        self.download = download
+        self.num_samples = num_samples
+        self.frame_zip_path = os.path.join(self.root, "frame.zip")
+        self.text_zip_path = os.path.join(self.root, "text.zip")
+        self.img_path = os.path.join(self.root, "JPEGImages")
+        self.box_path = os.path.join(self.root, "box/")
+        self.frame_path = os.path.join(self.root, "frame/")
+
+        # Check if Davis is installed and download it if necessary
+        self._check_directories()
+
+        self.exclude = [
+            "01-Light_video00016",
+            "01-Light_video00022",
+            "01-Light_video00023",
+            "02-SurfaceCover_video00012",
+            "03-Specularity_video00003",
+            "03-Specularity_video00012",
+            "10-LowContrast_video00013",
+        ]
+        self.input_size = input_size
+        self.transform = transform
+        self.x, self.y = self._parse_data(self.frame_path, self.box_path)
+        self.len = len(self.y)
+
+    def __len__(self):
+        return self.len
+
+    def __getitem__(self, idx):
+        sample, _ = self.get_sample(idx)
+        if self.transform:
+            sample = self.transform(sample)
+        return sample
+
+    def _parse_data(self, root, target_dir):
+        """
+        Parses ALOV dataset and builds tuples of (template, search region)
+        tuples from consecutive annotated frames.
+        """
+        x = []
+        y = []
+        envs = os.listdir(target_dir)
+        num_anno = 0
+        print("Parsing ALOV dataset...")
+        for env in envs:
+            env_videos = os.listdir(root + env)
+            for vid in env_videos:
+                if vid in self.exclude:
+                    continue
+                vid_src = self.frame_path + env + "/" + vid
+                vid_ann = self.box_path + env + "/" + vid + ".ann"
```

Comment:
Consider using an f-string instead, as-is this is creating 4 different strings. (`vid + ".ann"` creates a new one which is then added to `"/"` which creates a new one, and so forth)

I bet most interpreters can probably optimize it, but figure I'd mention it.

Reply:
good catch! Yeah I much prefer f string, surprised I wrote it this way